docs: add instructions for testing a PR by installing a downloaded binary in the $PATH (#5451)
ddev/ddev: docs/content/developers/building-contributing.md

Download and unzip the appropriate binary and place it in your `$PATH`.

> ### Homebrew with macOS

Reviewer: ```suggestion
### Homebrew with macOS or Linux
```

brew link --force ddev
```

> ### Installing a downloaded binary in the $PATH

Reviewer 2: We’ve almost got consistent title casing for headings on this page, so might as well continue here. And wrap `$PATH` in backticks since it’s a literal reference to a system variable. (Even though backtick styling in headings is kind of ugly.)

```suggestion
### Installing a Downloaded Binary in the `$PATH`
```

> Start by removing your current binary, for Debian flavor systems use this:

Reviewer 3: I want to add that normally you don't need to remove ddev because when you put any binary in `~/.local/bin` it takes precedence.

Reviewer: My approach is to use a symlink in /usr/local/bin/ddev, which points to either the system-installed ddev or the locally built one.

Author: We could add the symlink option as well? The reason I prefer `~/.local/bin` is that symlinks can be harder to understand.
Which commands would you use for the symlink solution?

Reviewer: I use symlinking. I switch back and forth with a script I call relink_ddev.sh. Of course I have to do this all the time. My strategy is to have a symlink /usr/local/bin/ddev.link that points to the built version and then have /usr/local/bin/ddev be a symlink that points to it. But delete /usr/local/bin/ddev when using system-installed version. I use this script on Linux/WSL2 as well as macOS.

```
#!/bin/bash

set -x
set -o errexit

if command -v brew; then
  brew unlink ddev || true
fi

SUDO=""
if  [ ! -w /usr/local/bin ] || [ ! -f /usr/local/bin/ddev ] || [ ! -w /usr/local/bin/ddev ]; then
  SUDO="sudo "
fi
${SUDO} ln -sf /usr/local/bin/ddev.link /usr/local/bin/ddev
hash -r
```

Author: Cool method @rfay, thanks for sharing. We could include it as an alternative method, if you suggest the changes (?), I can accept the suggestion and commit it to the patch.

Author: Yes, you're probably right. At least, it's now documented in this issue, if anyone is interested. Resolving this thread, since it seems to be so.

```
sudo apt remove ddev
```

> Next, unzip the binary you downloaded, make it executable, and move it to a folder in your path. Check with `echo $PATH`:

Reviewer 3: Not every distro has `~/.local/bin` in `$PATH`, so you can add an instruction how to do that (e.g. `export PATH="$HOME/.local/bin:$PATH"` to `~/.bashrc`).

Author: Great suggestion, I tried to update the text with that tip.

> Normally, you can put any binary in your path, and it takes precedence, so you don't need to remove or disable an already installed DDEV instance, which we will use here. In this example, `~/.local/bin` is used. `echo $PATH` and `which ddev` are valuable commands for debugging. Since not every distro has `~/.local/bin` in `$PATH`, you can create the folder and add it to your path in `~/.bashrc` with these commands:

Reviewer 2: Being picky, but I’d try to swap “is used” for more active language. Also helps break up a long stretch of comma clauses:

```suggestion
Normally, you can put any binary in your path, and it takes precedence, so you don't need to remove or disable an already installed DDEV instance, which we will use here. This example uses `~/.local/bin`. `echo $PATH` and `which ddev` are valuable commands for debugging. Since not every distro has `~/.local/bin` in `$PATH`, you can create the folder and add it to your path in `~/.bashrc` with these commands:
```

```
unzip ddev.zip
chmod +x ddev && mv ddev ~/.local/bin/ddev
```

> Verify the replacement worked by running `ddev version`. The output should be something like `DDEV version v1.22.3-39-gfbb878843`, instead of the regular `DDEV version v1.22.3`.

Reviewer 3: Also we can check the correct binary with `which ddev` and usually you need `source ~/.bashrc` to see the changes, but I usually just restart the terminal.

Author: Perfect, added.

> You need to restart DDEV, to download the Docker images that it needs

Reviewer 3: Dot at the end is missing.

Author: Done
